server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java

```diff
@@ -0,0 +1,642 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.api;
+
+import static org.apache.james.queue.api.MailQueueFixture.NAME1;
+import static org.apache.james.queue.api.MailQueueFixture.NAME2;
+import static org.apache.james.queue.api.MailQueueFixture.NAME3;
+import static org.apache.james.queue.api.MailQueueFixture.createMimeMessage;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Date;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.MailAddressFixture;
+import org.apache.mailet.base.test.FakeMail;
+import org.junit.jupiter.api.Test;
+
+public interface ManageableMailQueueContract {
+
+    ManageableMailQueue getManageableMailQueue();
+
+    @Test
+    default void getSizeShouldReturnZeroWhenNoMessage() throws Exception {
+        long size = getManageableMailQueue().getSize();
+
+        assertThat(size).isEqualTo(0L);
+    }
+
+    @Test
+    default void getSizeShouldReturnMessageCount() throws Exception {
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.OTHER_AT_LOCAL)
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+
+        long size = getManageableMailQueue().getSize();
+
+        assertThat(size).isEqualTo(1L);
+    }
+
+    @Test
+    default void dequeueShouldDecreaseQueueSize() throws Exception {
+        getManageableMailQueue().enQueue(FakeMail.builder()
+            .mimeMessage(createMimeMessage())
+            .sender(MailAddressFixture.OTHER_AT_LOCAL)
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .lastUpdated(new Date())
+            .build());
+
+        getManageableMailQueue().deQueue().done(true);
+
+        long size = getManageableMailQueue().getSize();
+
+        assertThat(size).isEqualTo(0L);
+    }
+
+    @Test
+    default void nackShouldNotDecreaseSize() throws Exception {
```

Comment:
nack? I guess no Ack but it's not very clear IMHO

Reply:
I'll improve it.